### PR TITLE
[IMP] Prevent to add both same journals on a period

### PR DESCRIPTION
**Pull request by adrienpeiffer**
Prevent to add both same journals on a period

### CODE REVIEW

File: account_journal_period_close/model/account_journal_period.py [hunk 1]
```diff
@@ -72,6 +72,18 @@ def create(self, cr, uid, values, context=None):
                     .browse(cr, uid, values['period_id'], context=context)
                 values.update({'name': (journal.code or journal.name)+':' +
                                (period.name or '')}),
+        if values.get('period_id') and values.get('journal_id'):
```
Comment (pedrobaeza) [inline account_journal_period_close/model/account_journal_period.py]: Better to be put as a constraint

Comment (adrienpeiffer) [inline account_journal_period_close/model/account_journal_period.py]: I think the best way is to create a sql unique constraint. @pedrobaeza do you agree with that ?

Comment (pedrobaeza) [inline account_journal_period_close/model/account_journal_period.py]: Yeah, that's it!

Comment (adrienpeiffer) [inline account_journal_period_close/model/account_journal_period.py]: Thank's

File: account_journal_period_close/model/account_journal_period.py [hunk 2]
```diff
@@ -41,6 +41,11 @@ class AccountJournalPeriod(orm.Model):
                                store=True, readonly=True)
     }
 
+    _sql_constraints = [
+        ('journal_period_uniq', 'unique(period_id, journal_id)',
+         'You can not add both same journal in a period.'),
```
Comment (pedrobaeza) [inline account_journal_period_close/model/account_journal_period.py]: Better to rephrase as `'You can add same journal in the same period twice.'`

File: account_journal_period_close/tests/test_account_journal_period_close.py [hunk 3]
```diff
@@ -194,3 +195,14 @@ def test_draft_move_close_journal(self):
                           jour_per_obj.action_done,
                           self.cr, self.uid, journal_period_ids,
                           context=context)
+
+    def test_duplicate_journal_period(self):
```
Comment (pedrobaeza) [inline account_journal_period_close/tests/test_account_journal_period_close.py:199]: Good having a test!

**Comment by pedrobaeza** [comment]
:+1: 

Thanks.

**Comment by sbidoul** [comment]
:+1: LGTM

**Comment by sbidoul** [comment]
As discussed offline, update the period form view to: 
- add a button to "Add all journals"
- make the list of journals read only when the period is closed (so a journal cannot be reopened when the period is closed)

**Comment by nbessi** [comment]
:+1:

**Pull request merged**